UnityProject/Assets/Scripts/Shuttles/EscapeShuttleConsole.cs

```diff
@@ -72,17 +83,70 @@ private void FinishHack(HandApply interaction)
 
 			beenEmagged = true;
 
-			if (GameManager.Instance.ShuttleSent) return;
+			DepartShuttle();
+		}
+
+		private void RegisterEarlyShuttleLaunch(IDCard card, PlayerScript performer)
+		{
+			if (GameManager.Instance.ShuttleSent)
+			{
+				Chat.AddExamineMsg(performer.gameObject, "The shuttle is already moving!");
+				return;
+			}
+			if (registeredIDs.Contains(card))
+			{
+				Chat.AddExamineMsg(performer.gameObject, "You've already done this!");
+				return;
+			}
+			registeredIDs.Add(card);
+
+			if (GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta && registeredIDs.Count >= 2)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			if (registeredIDs.Count >= 4)
+			{
+				DepartShuttle();
+				return;
+			}
+
+			AnnounceRemainingSwipesRequired();
+		}
+
+		private void AnnounceRemainingSwipesRequired()
+		{
+			var swpiesRequired = GameManager.Instance.CentComm.CurrentAlertLevel is CentComm.AlertLevel.Red or CentComm.AlertLevel.Delta ? 2 : 4;
+			var remainingSwipes = swpiesRequired - registeredIDs.Count;
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch has been request! need {remainingSwipes} more votes.</b></size></color>\n\n",
+				MatrixManager.MainStationMatrix);
+
+			Chat.AddSystemMsgToChat($"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch has been request! need {remainingSwipes} more votes.</b></size></color>\n\n",
```

Comment:
The text can be templated once and assigned to a variable.

Reply:
`$"\n\n<color=#FF151F><size={ChatTemplates.LargeText}><b>Escape Shuttle Emergency Launch has been request! need {remainingSwipes} more votes.</b></size></color>\n\n` is duplicated. It can be assigned to a variable and then used in both calls.